CONTRIBUTING.md

```diff
@@ -417,30 +264,168 @@ For services, click on the module name to reveal more details, then copy the nam
 >
 > Example: Searching for Oku (web browser) might also return Okular (document viewver), which share a similar names, but which are totally unrelated.
 
-## Exposing an NGI project
+## Adding a package
+
+1. To add a package, start by creating a `package.nix` file in the package directory `pkgs/by-name/some-package`, where `some-package` will be the package attribute name.
+
+   ```ShellSession
+   mkdir -p pkgs/by-name/some-package
+   $EDITOR pkgs/by-name/some-package/package.nix
+   ```
 
-In order to display a project on <ngi.nixos.org>, its metadata must be added to this repository's source code in a certain format.
+   Make sure to:
 
-1. Copy the [project template](./maintainers/templates/project) to the projects directory:
+   - Test the package on `x86_64-linux`.
 
+     ```ShellSession
+     git add pkgs/by-name/some-package
+     nix build .#some-package
+     ```
+
+   - Format the Nix expressions with [nix fmt](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-fmt.html).
+
+     ```ShellSession
+     nix fmt pkgs/by-name/some-package
```

Comment:
How about just using `nix fmt`? this way we know everything is correctly formatted, not just this particular path.

Reply:
I think it is better to provide the option to format only files which you changed to avoid contributors adding unrelated formatting changes.

Reply:
Done